test/e2e/specs/specs-wpcom/wp-editor__schedule.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * @group gutenberg
+ */
+
+import {
+	DataHelper,
+	TestAccount,
+	envVariables,
+	GutenbergEditorPage,
+	EditorSettingsSidebarComponent,
+	PublishedPostPage,
+} from '@automattic/calypso-e2e';
+import { Browser, Page } from 'playwright';
+
+declare const browser: Browser;
+
+describe( DataHelper.createSuiteTitle( `Editor: Schedule` ), function () {
+	const accountName = envVariables.GUTENBERG_EDGE
+		? 'gutenbergSimpleSiteEdgeUser'
+		: 'simpleSitePersonalPlanUser';
+	const postTitle = `Scheduled Post: ${ DataHelper.getTimestamp() }`;
+	const postContent = DataHelper.getRandomPhrase();
+	let postURL: string;
+	let gutenbergEditorPage: GutenbergEditorPage;
+	let page: Page;
+
+	beforeAll( async function () {
+		page = await browser.newPage();
+
+		const testAccount = new TestAccount( accountName );
+		await testAccount.authenticate( page );
+	} );
+
+	it( 'Go to the new post page', async function () {
+		gutenbergEditorPage = new GutenbergEditorPage( page );
+		await gutenbergEditorPage.visit( 'post' );
+	} );
+
+	it( 'Enter page title', async function () {
+		gutenbergEditorPage = new GutenbergEditorPage( page );
+		await gutenbergEditorPage.enterTitle( postTitle );
+	} );
+
+	it( 'Enter page content', async function () {
+		await gutenbergEditorPage.enterText( postContent );
+	} );
+
+	describe( 'Schedule: future', function () {
+		let editorSettingsSidebarComponent: EditorSettingsSidebarComponent;
+
+		it( 'Open scheduler', async function () {
+			const frame = await gutenbergEditorPage.getEditorFrame();
+			await gutenbergEditorPage.openSettings();
+			editorSettingsSidebarComponent = new EditorSettingsSidebarComponent( frame, page );
+			await editorSettingsSidebarComponent.clickTab( 'Post' );
+			await editorSettingsSidebarComponent.openSchedule();
+		} );
+
+		it( 'Schedule the post for next year', async function () {
+			const currentDate = new Date();
+			await editorSettingsSidebarComponent.schedule( {
+				year: currentDate.getFullYear() + 1,
```

Comment:
Very small nit, but it took me a couple read throughs to make sure I understood all the date logic here. I liked the approach you did in the past section where you took a date and modified it to be what you wanted, and then called the schedule function with the corresponding parts. What do you think of doing that here too for parity?

So, e.g.
```typescript
const futureDate = new Date();
futureDate.setFullYear( new Date().getFullYear() + 1);
```

Reply:
I've modified this part to first define the date object, then use the date object to set the params.